server/src/main/java/com/objectcomputing/checkins/services/checkins/CheckInController.java

```diff
@@ -47,17 +58,27 @@
                 .body(error);
     }
 
-
     /**
      * Find Check-in details by Member Id or PDL Id. 
      * @param teamMemberId
      * @param pdlId
      * @return
      */
     @Get("/{?teamMemberId,pdlId,completed}")
-    public Set<CheckIn> findByValue(@Nullable UUID teamMemberId,
-                                     @Nullable UUID  pdlId, @Nullable Boolean completed) {          
-            return checkInservices.findByFields(teamMemberId, pdlId,completed);
+    public Set<CheckIn> findByValue(@Nullable UUID teamMemberId, @Nullable UUID  pdlId, @Nullable Boolean completed) {
```

Comment:
These validations are expensive since DB call is made prior to validating the access of the current user. Could anyone think of a better way to do this?

Reply:
The @Secured annotation, when used on a method, can be used to specify a role.
https://micronaut-projects.github.io/micronaut-security/latest/guide/#secured

Reply:
that will not help in this situation, right ? Because we are validating if a current user is a PDL of the subject that he is doing a check in and not if he has a PDL role because many people will have PDL role but need to make sure they do the checkin for their team member.

Reply:
It will replace the securityService call.

As for the rest there is another option but whether it is better or not is debatable.

We can add claims to our user details mapper that will contain the user's PDL's ID and (if the user is a PDL) any team members for whom the user is a PDL. You would be able to check that against the request body instead of having to go through the user service.

Reply:
No, it won't help us here. This validation has to be done by hand.

Reply:
I made a branch, [here](https://github.com/oci-labs/check-ins/tree/feature-245/also-with-claim-validation), which has a really hacky way to do that. If I understand what y'all are trying to do we should be able to use something like that.

Reply:
I'm actually not against the idea of keeping a user's assigned team members in their user details...but I think we should make a separate story to investigate that.